intentgen/src/main/java/com/chatbot/IntentGenerator.java

```diff
@@ -0,0 +1,190 @@
+        .setText(Intent.Message.Text.newBuilder().addAllText(intent.getResponsesList()).build())
+        .build();
+    return intentProtobufBuilder.addMessages(message);
+  }
+
+  private static Intent.Builder addOutputContexts(UseCase.Intent intent,
+      Intent.Builder intentProtobufBuilder, String projectID) {
+    List<Context> outputContextProtobufList= new ArrayList<Context>();
+    for(String successor: intent.getSuccessorsList()) {
+      Context contextProtobuf = Context.newBuilder()
+      .setLifespanCount(1)
+      .setName("projects/" + projectID + "/agent/sessions/-/contexts/" + successor + "Context")
+      .build();
+      outputContextProtobufList.add(contextProtobuf);
+    }
+    return intentProtobufBuilder.addAllOutputContexts(outputContextProtobufList);
+  }
+
+  private static void createOrUpdateIntents(List<Intent> intentProtobufList,
+    IntentsClient intentsClient, Map<String, String> intentDisplayNameToName,
+    ProjectAgentName parent) {
+    for(Intent intent: intentProtobufList) {
+      if(intentDisplayNameToName.containsKey(intent.getDisplayName())) {
+        intentsClient.updateIntent(intent, "en");
+      }
+      else {
+        intentsClient.createIntent(parent, intent);
+      }
+    }
+  }
+
+  private static Intent intentToIntentProtobuf(UseCase.Intent intent,
+    Map<String, String> intentDisplayNameToName, String projectID) {
+    Intent.Builder intentProtobufBuilder = Intent.newBuilder();
+    intentProtobufBuilder.setDisplayName(intent.getIntentName());        
+    // output contexts need to be set to the input contexts of the successor intents
+    addOutputContexts(intent, intentProtobufBuilder, projectID);
+    intentProtobufBuilder.addInputContextNames("projects/" + projectID +
+        "/agent/sessions/-/contexts/" + intent.getIntentName() + "Context");
+
+    intentProtobufBuilder = addTrainingPhrases(intent, intentProtobufBuilder);
+    intentProtobufBuilder = addResponses(intent, intentProtobufBuilder);        
+    intentProtobufBuilder.addAllEvents(intent.getEventsList());
+
+    if(intentDisplayNameToName.containsKey(intent.getIntentName())) {
+      intentProtobufBuilder.setName(intentDisplayNameToName.get(intent.getIntentName()));
+    }
+    return intentProtobufBuilder.build();
+  }
+
+  private static Options defineOptions() {
+    Options options = new Options();
+    final Option useCaseFile = Option.builder("f").hasArg()
+        .desc("prototxt file describing the use case" ).build();
+    options.addOption(useCaseFile);
+    return options;
+  }
+  
+  private static String getUseCaseFile(String [] args) throws ParseException {
+    final Options options = defineOptions();
+    final CommandLineParser parser = new DefaultParser();
+    final CommandLine cmd = parser.parse(options, args);
+    final String fileName = cmd.getOptionValue("f");
+    if(fileName.equals("")) {
+      throw new ParseException("No file name provided");
+    }
+    return fileName;
+  }
```

Comment:
This function can only be used within this main. As such it doesn't serve much purpose. Create functions when Parts of code could be reused in the same code or outside. This might be useful here if there were multiple args to fetch in which case the name would be like getArgsByOptionValue. Even there it might be an issue as return type could be different.

Reply:
Will move it to main